v4/export/writer_util.go

```diff
@@ -189,6 +193,19 @@ func WriteInsert(pCtx context.Context, tblIR TableDataIR, w storage.Writer, file
 			}
 			counter += 1
 			wp.AddFileSize(uint64(bf.Len()-lastBfSize) + 2) // 2 is for ",\n" and ";\n"
+			failpoint.Inject("ChaosBrokenMySQLConn", func(val failpoint.Value) {
+				brokenPoint := val.(int)
+				if counter >= brokenPoint {
+					var err error
+					doOnce.Do(func() {
+						tblIR.Close()
+						err = errors.New("connection is closed")
+					})
```

Comment:
why use `counter >= brokenPoint` with a `sync.Once` instead of checking `counter == brokenPoint`, or [use failpoint's `prob%count*action` feature](https://github.com/pingcap/failpoint/#key-concepts)

```sh
export GO_FAILPOINTS="github.com/pingcap/dumpling/v4/export/ChaosBrokenMySQLConn=1%1*return(1)" 
```